.github/scripts/ext-debian.sh

```diff
@@ -44,9 +44,21 @@ stage_source() {
   debian_package_name=$(dpkg-parsechangelog --show-field Source)
   full_version=$(dpkg-parsechangelog --show-field Version)
   debian_version="${full_version%-*}"
+
+  # Special case for creating source tarball of a Rust package vendored dependencies to allow packages with unpackage deps in debian
+  # See https://blog.shadura.me/2020/12/22/vendoring-rust-in-debian-derivative/
+  if [ -f debian/cargo-checksum.json ]; then # Assume cargo-checksum.json existing implies needing to generate vendored deps
+    echo -e "\033[0;34mGenerating vendor source tarball from cargo\033[0m"
+    [ -d vendor ] && rm -rf vendor
+    rm -rf Cargo.lock || true
+    cargo vendor
+    tar Jcf ../${debian_package_name}_${debian_version}.orig-vendor.tar.xz vendor/
```

Comment:
I see, in the blog post, that new stanza is added in `rules`, but I don't get how does this tarball being used? Do we also need to push this tarball to deb archive?

Reply:
This feature requires an integration on the package side.  In the build script of a rust package we'd like to "autovendor", we update the cargo configuration to include the packages.  See here for the example that was tested:

https://github.com/kgilmer/elbey/pull/12/files#diff-889c5495b68d2a0c0d1f238818716a71939918499efbc0fc6d2994319624fc67R13

Reply:
I added some basic docs, please LMK if there is still ambiguity or questions